### PR TITLE
Add support for device identity operations

### PR DESCRIPTION
**Pull request by corollaries**
This PR adds support for getting, creating, updating, and deleting device identities from the IoT Hub and partial implementation for the module identities. I'll add full support for the module identities in a later PR.

### CODE REVIEW

File: sdk/iothub/src/service/identity.rs [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+pub struct X509ThumbPrint {
+    pub primary_thumbprint: Option<String>,
+    pub secondary_thumbprint: Option<String>,
+}
+
+/// AuthenticationType of a module or device
+#[derive(Serialize, Debug, Deserialize, PartialEq)]
+pub enum AuthenticationType {
+    #[serde(rename = "certificate")]
+    Certificate,
```
Comment (rylev) [inline sdk/iothub/src/service/identity.rs:57]: Nit: we should either add docs for each of the enum variants or `#[allow(missing_docs)]` for this enum. At some point we should turn on `!#[deny(missing_docs)]`.

File: sdk/iothub/src/service/identity.rs [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+impl<'a> DeviceIdentityBuilder<'a> {
+    pub(crate) fn new(
+        service_client: &'a ServiceClient,
+        operation: IdentityOperation,
+        etag: Option<String>,
+    ) -> Self {
+        Self {
+            service_client,
+            authentication: None,
+            capabilities: DeviceCapabilities { iotedge: false },
```
Comment (rylev) [inline sdk/iothub/src/service/identity.rs:140]: nit: maybe add a `#[derive(Default)]` to `DeviceCapabilities`?

File: sdk/iothub/src/service/mod.rs [hunk 3]
```diff
@@ -403,6 +407,105 @@ impl ServiceClient {
...
+    ///
+    /// ```
+    /// use iothub::service::ServiceClient;
+    /// use serde_json;
+    ///
+    /// # let connection_string = "HostName=cool-iot-hub.azure-devices.net;SharedAccessKeyName=iothubowner;SharedAccessKey=YSB2ZXJ5IHNlY3VyZSBrZXkgaXMgaW1wb3J0YW50Cg==";
+    /// let iothub = ServiceClient::from_connection_string(connection_string, 3600).expect("Failed to create the ServiceClient!");
+    /// let device = iothub.get_device_identity("some-device");
+    /// ```
+    pub async fn get_device_identity<S>(&self, device_id: S) -> Result<Device, AzureError>
```
Comment (rylev) [inline sdk/iothub/src/service/mod.rs:421]: These don't really match the pattern of request builders we've been using in other crates. That might be fine, but we probably need to think more deeply about if the iothub crate should follow the same pattern.

**Review by rylev — APPROVED** [review]
Looks good. Some nits and some food for thought, but nothing that should prevent us from merging this.

File: sdk/iothub/src/service/mod.rs (hunk 3 above)
Comment (corollaries) [reply sdk/iothub/src/service/mod.rs:421]: I'd be happy to change it so it's similar to the other crates, no problem at all. I did notice there is a "common" http client within the core crate that was added a while ago. Is this what you mean that should be used instead? Could you link me an example of the ideal way we'd be doing requests?

File: sdk/iothub/src/service/identity.rs (hunk 1 above)
Comment (corollaries) [reply sdk/iothub/src/service/identity.rs:66]: Thanks, I'll go over the rest of the `iothub` crate and add any missing docs and enable `#![deny(missing_docs)]` in a future PR.

**Review by rylev — APPROVED** [review]
Great stuff!

**Pull request merged**